policy/executor/graph.go

```diff
@@ -18,16 +18,16 @@ type GraphExecutor interface {
 }
 
 func ExecuteResolvedPolicy(schema *resources.Schema, runtime *resources.Runtime, collectorSvc policy.PolicyResolver, asset *asset.Asset,
-	resolvedPolicy *policy.ResolvedPolicy, features cnquery.Features, progressProg progress.Program,
+	resolvedPolicy *policy.ResolvedPolicy, features cnquery.Features, multiProgressBar progress.MultiProgress,
```

Comment:
I might be missing something here, but why does the execution need to care that its a multi progress interface. It can only execute for one thing. Something else that cares about displaying multiple progress bars can deal with doing whatever necessary transform to make it work with multiple progress bars. For example:

```
type Progress interface {
  OnProgress(percent float64)
}

type MultiProgressAdapter struct {
  multi MultiProgress
  key string
}

func (m* MultiProgressAdapter) OnProgress(percentage float64) {
  m.multi.OnProgress(key, percentage)
} 

for _, asset := range assets {
  p := MultiProgressAdapter{key: asset.PlatformIds[0], multi: multiProgressBar)
  ExecuteResolvedPolicy(asset.Mrn, ..., p,...)
}
```

Reply:
Thanks for explaining this to me. I re-worked the cnquery and cnspec PRs to re-use the extended progress.Progress interface.